detox/e2e/test/custom_statuses/custom_status_expiry.e2e.js

```diff
@@ -0,0 +1,214 @@
+        await expect(element(by.text('Until ' + expiryText))).toBeVisible();
+
+        await ChannelInfoScreen.close();
+    });
+
+    it("MM-T4092 RN apps: Custom Status Expiry - Editing 'Clear After' Time (mobile)", async () => {
+        // # Open custom status screen
+        await openSettingsSidebar();
+        await CustomStatusScreen.open();
+
+        // * Check if all the default suggestions are visible
+        const isSuggestionPresentPromiseArray = [];
+        defaultCustomStatuses.map(async (text) => {
+            isSuggestionPresentPromiseArray.push(expect(getCustomStatusSuggestion(text)).toBeVisible());
+        });
+        await Promise.all(isSuggestionPresentPromiseArray);
+
+        // * Tap a suggestion and check if it is selected
+        await tapSuggestion(defaultStatus);
+        await expect(element(by.id(`custom_status.duration.${defaultStatus.duration}`))).toBeVisible();
+
+        // # Click on the Clear After option and check if the modal opens
+        await ClearAfterScreen.open();
+
+        // # Select a different expiry time and check if it is shown in Clear After
+        await ClearAfterScreen.tapSuggestion('four_hours');
+        await ClearAfterScreen.close();
+
+        // Check if selected duration is shown in clear after
+        await expect(element(by.id('custom_status.duration.four_hours'))).toBeVisible();
+
+        // * Open Clear After modal
+        await ClearAfterScreen.open();
+
+        // * Tap 'Custom' and check if it is selected
+        await element(by.text('Custom')).tap();
+
+        // * Check if select date and select time buttons are visible
+        expect(element(by.id(ClearAfterScreen.testID.selectDateButton))).toBeVisible();
+        expect(element(by.id(ClearAfterScreen.testID.selectTimeButton))).toBeVisible();
+
+        const am_pm = moment().format('A').toString();
+
+        // * Select some time in future in date time picker
+        await ClearAfterScreen.openTimePicker();
+
+        if (isAndroid()) {
+            await DateTimePicker.changeTimeAndroid('11', '30');
+            await DateTimePicker.tapOkButtonAndroid();
+        } else {
+            const timePicker = DateTimePicker.getDateTimePickerIOS();
+            await timePicker.setColumnToValue(0, '11');
+            await timePicker.setColumnToValue(1, '30');
+        }
+
+        await ClearAfterScreen.close();
+        await CustomStatusScreen.close();
+
+        // * Check if the selected emoji and text are visible in the sidebar
+        await openSettingsSidebar();
+        await expect(element(by.text(defaultStatus.text))).toBeVisible();
+        await expect(element(by.id(`custom_status.emoji.${defaultStatus.emoji}`))).toBeVisible();
+        await expect(element(by.text('(Until 11:30 ' + am_pm + ')'))).toBeVisible();
+
+        // # Close settings sidebar
+        await closeSettingsSidebar();
+    });
```

Comment:
Ran and passed both in iOS and Android.
<img width="951" alt="Screen Shot 2021-07-22 at 7 41 18 AM" src="https://user-images.githubusercontent.com/5334504/126577971-a10fdbc0-6398-4f01-9305-7ea5070eb613.png">
<img width="951" alt="Screen Shot 2021-07-22 at 8 15 07 AM" src="https://user-images.githubusercontent.com/5334504/126577974-6047acad-f8dd-47db-93fb-fd71f55a5724.png">